### PR TITLE
Fix resolver returning null when record has service objects or map of service objects as its field

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Purpose

Fixes: [#3466](https://github.com/ballerina-platform/ballerina-standard-library/issues/3466)
Fixes: [#3579](https://github.com/ballerina-platform/ballerina-standard-library/issues/3579)

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [ ] ~~Updated the spec~~
- [ ] ~~Checked native-image compatibility~~

### CODE REVIEW

File: ballerina-tests/tests/test_services.bal [hunk 1]
```diff
@@ -1717,3 +1717,31 @@ service /covid19 on basicListener {
...
     }
 }
+
+table<Review> reviews = table [
+        {product: new ("1"), score: 20, description: "Product 01"},
+        {product: new ("2"), score: 20, description: "Product 02"},
+        {product: new ("3"), score: 20, description: "Product 03"},
+        {product: new ("4"), score: 20, description: "Product 04"},
+        {product: new ("5"), score: 20, description: "Product 05"}
+    ];
```
Comment (DimuthuMadushan) [inline ballerina-tests/tests/test_services.bal:1727]:
```suggestion
table<Review> reviews = table [
    {product: new ("1"), score: 20, description: "Product 01"},
    {product: new ("2"), score: 20, description: "Product 02"},
    {product: new ("3"), score: 20, description: "Product 03"},
    {product: new ("4"), score: 20, description: "Product 04"},
    {product: new ("5"), score: 20, description: "Product 05"}
];
```

File: ballerina-tests/tests/07_service_objects.bal [hunk 2]
```diff
@@ -77,3 +77,46 @@ isolated function testComplexService() returns error? {
     };
     assertJsonValuesWithOrder(result, expectedPayload);
 }
+
+@test:Config {
+    groups: ["service", "records", "a"]
```
Comment (DimuthuMadushan) [inline ballerina-tests/tests/07_service_objects.bal:82]:
```suggestion
    groups: ["service", "records"]
```

Comment (MohamedSabthar) [reply ballerina-tests/tests/07_service_objects.bal]: Fixed with [`fbb3ab4`](https://github.com/ballerina-platform/module-ballerina-graphql/pull/1029/commits/fbb3ab46ce2d4b8a2981fa8c5ea27ccb02c3b813)

File: ballerina-tests/tests/test_services.bal (hunk 1 above)
Comment (MohamedSabthar) [reply ballerina-tests/tests/test_services.bal]: Fixed with [`fbb3ab4`](https://github.com/ballerina-platform/module-ballerina-graphql/pull/1029/commits/fbb3ab46ce2d4b8a2981fa8c5ea27ccb02c3b813)

**Review by ThisaruGuruge — COMMENTED** [review]
How does this handle the subscription scenarios? SInce the subscriptions are handled through a separate flor in 2201.2.x branch, we have to fix that.

**Comment by MohamedSabthar** [comment]
> How does this handle the subscription scenarios? SInce the subscriptions are handled through a separate flor in 2201.2.x branch, we have to fix that.

Yes I noticed subscription is handled in java level and this bug is not present in java implementation. I'll update the PR with test cases for subscription

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**